[18.0][MIG] purchase_order_archive: Migration to 18.0

BhaveshHeliconia opened the pull request.

astirpe committed:
[11.0][ADD] purchase_order_archive

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update purchase_order_archive.pot

astirpe committed:
[12.0][MIG] purchase_order_archive

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update purchase_order_archive.pot

OCA-git-bot committed:
[UPD] README.rst

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (3 of 3 strings)

Translation: purchase-workflow-12.0/purchase-workflow-12.0-purchase_order_archive
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-purchase_order_archive/zh_CN/

astirpe committed:
[13.0][MIG] purchase_order_archive

oca-travis committed:
[UPD] Update purchase_order_archive.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-13.0/purchase-workflow-13.0-purchase_order_archive
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-13-0/purchase-workflow-13-0-purchase_order_archive/

OCA-git-bot committed:
pre-commit update

dong-z committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (4 of 4 strings)

Translation: purchase-workflow-13.0/purchase-workflow-13.0-purchase_order_archive
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-13-0/purchase-workflow-13-0-purchase_order_archive/zh_CN/

astirpe committed:
[14.0][MIG] purchase_order_archive

oca-travis committed:
[UPD] Update purchase_order_archive.pot

OCA-git-bot committed:
[UPD] README.rst

victoralmau committed:
[IMP] purchase_order_archive: Improve filters (add to all search views + add active filter)

TT44770

Commit:
[UPD] Update purchase_order_archive.pot

OCA-git-bot committed:
purchase_order_archive 14.0.1.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-14.0/purchase-workflow-14.0-purchase_order_archive
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-14-0/purchase-workflow-14-0-purchase_order_archive/

OCA-git-bot committed:
[UPD] README.rst

CLaurelB committed:
[MIG] purchase_order_archive: Migration to 16.0

Commit:
[UPD] Update purchase_order_archive.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_order_archive
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_order_archive/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_order_archive
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_order_archive/it/

Commit:
[IMP] purchase_order_archive: pre-commit auto fixes

Commit:
[MIG] purchase_order_archive: Migration to 17.0

Commit:
[UPD] Update purchase_order_archive.pot

OCA-git-bot committed:
[BOT] post-merge updates

BhaveshHeliconia committed:
[MIG] purchase_order_archive: Migration to 18.0

StefanRijnhart reviewed (commented):
Not your fault, but this module seems to have a pretty bad flaw: you can only archive a cancelled or locked order, but once archived you can set the cancelled order to draft and confirm it etc. etc.
Would you consider adding a constraint that prevents archived orders in any other states than cancel or done? If you do, a simple test would be good to have as well.

BhaveshHeliconia commented:
@StefanRijnhart , Thank you for pointing this out and for the valuable suggestion! I've implemented a constraint to prevent archived orders from being in states other than "cancel" or "done." I’ve also added a simple test case to validate this behavior.

I’d greatly appreciate it if you could review the changes and let me know if there's anything else to improve.

Thank you again for your input!

StefanRijnhart reviewed (approved):
Great, thank you!

HviorForgeFlow reviewed (approved):
Code review :+1:

HviorForgeFlow commented:
/ocabot migration purchase_order_archive

HviorForgeFlow commented:
/ocabot merge nobump
